src/request.rs

```diff
@@ -11,11 +11,39 @@ use crate::{
     AData, ADataAddress, ADataAppend, ADataIndex, ADataOwner, ADataPubPermissions,
     ADataUnpubPermissions, ADataUser, AppPermissions, Coins, IDataAddress, IDataKind, MDataAddress,
     MDataPermissionSet, MDataSeqEntryActions, MDataUnseqEntryActions, PublicKey, SeqMutableData,
-    UnseqMutableData, XorName,
+    Signature, UnseqMutableData, XorName,
 };
 use serde::{Deserialize, Serialize};
 use std::fmt;
 
+pub type TransactionId = u64; // TODO: Use the trait UUID
+
+#[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
+pub struct AccountData {
+    destination: XorName,
+    authorised_getter: PublicKey, // deterministically created from passwords
+    data: Vec<u8>,                // up to 1 MB
+    signature: Signature,
+}
+
+impl AccountData {
```

Comment:
I probably should have mentioned in the last review, but with the fields being private, we need a constructor now I think.

It might also be worth adding a pub const `MAX_ACCOUNT_DATA_BYTES` and having something like `pub fn size_is_valid(&self) -> bool` where we do `self.data.len() <= MAX_ACCOUNT_DATA_BYTES`.

This could be called inside the constructor and return an error if it's false.

We could even add that check when deserialising and likewise return an error, then `size_is_valid` wouldn't even need to be `pub`.  We'd be guaranteed that every instance of `AccountData` is validly-sized.  However, implementing `Deserialize` by hand is maybe not worth the risk.